src/uterm_input_uxkb.c

```diff
@@ -380,19 +437,40 @@ int uxkb_dev_process(struct uterm_input_dev *dev,
 		     uint16_t key_state, uint16_t code)
 {
 	struct xkb_state *state;
+	struct xkb_compose_state *compose_state;
 	xkb_keycode_t keycode;
 	const xkb_keysym_t *keysyms;
+	xkb_keysym_t one_sym;
 	int num_keysyms, ret;
+	enum xkb_compose_status compose_status;
 	enum xkb_state_component changed;
 
 	if (key_state == KEY_REPEATED)
 		return -ENOKEY;
 
 	state = dev->state;
+	compose_state = dev->compose_state;
 	keycode = code + EVDEV_KEYCODE_OFFSET;
 
 	num_keysyms = xkb_state_key_get_syms(state, keycode, &keysyms);
 
```

Comment:
Please add a comment explaining the following logic. It's rather complex and for a person unfamiliar with the subject (like me), it takes a lot of reading and research to understand what's going on. So I think it's better to document these.

Something along these lines
```
    /*
     * To summarize the following convoluted logic:
     *
     * - single key press may produce one or more keysyms
     * - if num_keysyms == 1,
     *     + use get_one_sym to handle the maybe present LOCK mod
     *     + use the resulting one_sym to feed the compose_state
     *     + compose_state will always produce one keysym, which is set back to one_sym
     * - if num_keysyms != 1,
     *     + LOCK mod translation doesn't make sense, so skip that
     *     + compose_state is fed NoSymbol per documentation and the result is basically ignored
     * - update xkb state after querying keysyms per documentation
     * - if in the process of composing, stop here
     * - otherwise process events with keysyms and num_keysyms
     *
     * Some background on compose handling:
     * - https://github.com/xkbcommon/libxkbcommon/issues/4
     */
```